services/provider/server/server.go

```diff
@@ -741,17 +742,36 @@ func (s *OCSProviderServer) isSubscriptionChannelValid(ctx context.Context, subs
 	}
 
 	// Check if the installed CSV is the expected one for the channel from the package manifest
-	packageManifest := &unstructured.Unstructured{}
-	packageManifest.SetGroupVersionKind(schema.GroupVersionKind{
+	packageManifestList := &unstructured.UnstructuredList{}
+	packageManifestList.SetGroupVersionKind(schema.GroupVersionKind{
 		Group:   "packages.operators.coreos.com",
 		Version: "v1",
 		Kind:    "PackageManifest",
 	})
-	err = s.client.Get(ctx, client.ObjectKey{Name: subscription.Spec.Package, Namespace: s.namespace}, packageManifest)
+	err = s.client.List(ctx, packageManifestList,
+		client.InNamespace(s.namespace),
+		// Field match prone to future change, but better than processing hundreds of resources, consuming more CPU/memory
+		client.MatchingFields{"metadata.name": packageName},
+	)
```

Comment:
A small nit pick about formatting. If we are to split a function call into multiple lines we put each argument on its own line for reading consistency 
```suggestion
	err = s.client.List(
		ctx, 
		packageManifestList,
		client.InNamespace(s.namespace),
		// Field match prone to future change, but better than processing hundreds of resources, consuming more CPU/memory
		client.MatchingFields{"metadata.name": packageName},
	)
```